incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -413,6 +413,7 @@ func (r *HierarchyConfigReconciler) syncLabel(log logr.Logger, nsInst *corev1.Na
 		anc = anc.Parent()
 		depth++
 	}
+	ns.SetLabels(nsInst.Labels)
```

Comment:
This is probably worth a comment ("why are we doing this?")

Reply:
Agreed. Is it to update the labels in the forest?

Reply:
done

Reply:
This comment doesn't answer the question of _why_ we're doing this - it just restates _what_ the code is doing but in a different way. Usually, you can figure out the "what" just by reading the code more carefully, but "why" is much harder to figure out.

Imagine you're looking at this code in a year and you've completely forgotten why you wrote it. Can you leave yourself a hint so that you know what to look at to understand why you can't delete it?